### DISCUSSION TITLE
Insert Musical Notation

### DISCUSSION THREAD

**Discussion by protean-onion** [TriliumNext/Trilium#3976] [Plugin Ideas]
Hello,

I like playing music, and I often use Trilium to take notes about what I am playing and listening to. I thought it would be great if I could insert musical notation in my Trilium notes.

Would that be too much programming? If not, it would be great if that could be added.

Thanks for your time, and thanks for creating and maintaining Trilium. This is great. I hope you are well. Sending my regards.

**Comment by meichthys** [comment]
You MAY be able to use the excalidraw canvas note since that supports musical notation: https://github.com/excalidraw/excalidraw-libraries/pull/743

**Reply by zadam** [reply]
There's currently no support for excalidraw libraries. But I do want to add support for them.

**Comment by zadam** [comment]
Hello, sorry to disappoint, but I don't plan such support (apart from possibly excalidraw library mentioned above). It's a too specialized/niche feature.

**Comment by Kirtai** [comment]
Any possibility of adding this now via [abcjs](https://www.abcjs.net/) which is a javascript library for notation using the [ABC](https://abcnotation.com/) text standard?

**Reply by eliandoran** [reply]
It can theoretically be embedded via custom widgets. In the future we plan to have plugin support, including the ability to create custom note types. 

Since it's a usecase that is only useful for a subset of our users, we don't plan to have it as a core functionality.

**Reply by Kirtai** [reply]
Alright. I look forward to the addition of plugins :)

**Comment by acertaingrace** [comment]
It's not as integrated as I would like but using the abcjs library that @Kirtai mentioned with Render Notes works pretty much ok. Here's an example of using abcjs in Trilium ([x](https://github.com/acertaingrace/trilium-abcjs)). The biggest issue I'm facing is abcjs injecting an in-line style tag and somehow permanently affecting the html file.
